src/components/Vehicle.cxx

```diff
@@ -0,0 +1,117 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include <Urho3D/Core/Context.h>
+#include <Urho3D/Graphics/DebugRenderer.h>
+#include <Urho3D/Graphics/DecalSet.h>
+#include <Urho3D/Graphics/Material.h>
+#include <Urho3D/Graphics/Model.h>
+#include <Urho3D/Graphics/ParticleEffect.h>
+#include <Urho3D/Graphics/ParticleEmitter.h>
+#include <Urho3D/Graphics/StaticModel.h>
+#include <Urho3D/IO/Log.h>
+#include <Urho3D/Physics/CollisionShape.h>
+#include <Urho3D/Physics/Constraint.h>
+#include <Urho3D/Physics/PhysicsEvents.h>
+#include <Urho3D/Physics/PhysicsWorld.h>
+#include <Urho3D/Physics/RaycastVehicle.h>
+#include <Urho3D/Resource/ResourceCache.h>
+#include <Urho3D/Scene/Scene.h>
+#include "components/Vehicle.hxx"
+
+
+void Vehicle::RegisterObject(Urho3D::Context* context) {
+    context->RegisterFactory<Vehicle>();
+}
+
+Vehicle::Vehicle(Urho3D::Context* context) : LogicComponent(context) {
+}
+Vehicle::~Vehicle() = default;
+
+void Vehicle::Init(const smce::VehicleConfig& vconf) {
+    auto* vehicle = node_->CreateComponent<Urho3D::RaycastVehicle>();
+    vehicle->Init();
+    auto* hullBody = node_->GetComponent<Urho3D::RigidBody>();
+    hullBody->SetMass(1.33f);
+    hullBody->SetLinearDamping(0.2f);
+    hullBody->SetAngularDamping(0.5f);
+    hullBody->SetCollisionLayer(1);
+    auto* cache = GetSubsystem<Urho3D::ResourceCache>();
+    auto* hullObject = node_->CreateComponent<Urho3D::StaticModel>();
+    auto* hullColShape = node_->CreateComponent<Urho3D::CollisionShape>();
+    Urho3D::Vector3 v3BoxExtents = Urho3D::Vector3::ONE;
+    hullColShape->SetBox(v3BoxExtents);
+    node_->SetScale(Urho3D::Vector3(0.195f, 0.0666f, 0.236f)); /// Chassi 195mm*66.6mm*236mm
+    hullObject->SetModel(cache->GetResource<Urho3D::Model>(vconf.hull_model_file.generic_string().c_str()));
+    hullObject->SetMaterial(cache->GetResource<Urho3D::Material>("Materials/offroadVehicle.xml"));
+    hullObject->SetCastShadows(true);
+    Urho3D::Vector3 wheelDirection(0, -1, 0);
+    Urho3D::Vector3 wheelAxle(-1, 0, 0);
+    for (auto& i = vconf.parts.begin(); i != vconf.parts.end(); i++) {
```

Comment:
this should be a [range-based loop](https://en.cppreference.com/w/cpp/language/range-for)